test/test_euler.py

```diff
@@ -0,0 +1,781 @@
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+import numpy.linalg as la  # noqa
+import pyopencl as cl
+import pyopencl.clrandom
+import pyopencl.clmath
+import logging
+import pytest
+
+from pytools.obj_array import (
+    flat_obj_array,
+    make_obj_array,
+)
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from grudge.symbolic.primitives import TracePair
+from mirgecom.euler import inviscid_operator
+from mirgecom.euler import split_conserved
+from mirgecom.steppers import euler_flow_stepper
+from mirgecom.initializers import Vortex2D
+from mirgecom.initializers import Lump
+from mirgecom.boundary import PrescribedBoundary
+from mirgecom.boundary import DummyBoundary
+from mirgecom.eos import IdealSingleGas
+from grudge.eager import EagerDGDiscretization
+from pyopencl.tools import (  # noqa
+    pytest_generate_tests_for_pyopencl as pytest_generate_tests,
+)
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+def test_inviscid_flux(ctx_factory, dim):
+    """Checks that the Euler-internal inviscid flux
+    routine (_inviscid_flux) returns exactly the
+    expected result. The test checks flux for 1,
+    2, and 3 spatial dimensions.
+
+    The test is in 3 parts:
+
+    - Exact expression: Checks the returned flux
+      against the exact expressions:
+      F(q) = <rhoV, (E+p)V, rho(V.x.V) + pI>
+    - Prescribed p, V = 0:  Checks that only
+      diagonal terms of the momentum flux
+      [ rho(V.x.V) + pI ] are non-zero and return
+      the correctly calculated p.
+    - Prescribed p, V != 0: Checks that the
+      flux terms are returned in the proper
+      order by running only 1 non-zero velocity
+      component at-a-time.
```

Comment:
Would it make sense to make these three separate tests? (i.e. how much are the tests really benefiting from being in the same monster function?)

Reply:
Yes, makes sense.

Reply:
I was trying to find a convenient way to attach commits to certain comments.  Is there some automated mechanism for that?

Reply:
I know exactly what you mean, and no, I don't think such a tool exists, unfortunately.

Reply:
Unfortunate indeed.  It would be great if only we had something like this:
https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#about-linked-issues-and-pull-requests

.... with a keyword construction in the commit message that would help us link commits to certain discussions and/or review comments.  

Alas..